### DISCUSSION TITLE
Edit Response

### DISCUSSION THREAD

**Discussion by BctfN0HUK7Yg** [emilk/egui#981] [Q&A]
I want to change Response fileds (like clicked or other) in my component. How I can do it?
--------
 if response.clicked() {
    do();
    response.clicked= false; // ?????
}

**Comment by emilk** [comment]
You can't. `Response` is immutable. Are you sure you need to do this? What is the context?

**Reply by BctfN0HUK7Yg** [reply]
I'm still trying to make recursive widget in widget in widget and get the response of selected one.
I have found one way to make it, but it's ugly. 
I use struct MyResponse { element_id, Response} and put reference to my widget.

In this question I try to change a widget response when inner widget has response, so I need to mutate response or shadow it with empty response;

May be you know better way to do it?

**Reply by emilk** [reply]
hard to say without a simple reproduce/example of what you're attempting
